actionpack/lib/action_dispatch/middleware/host_authorization.rb

```diff
@@ -17,7 +17,17 @@ module ActionDispatch
   # if +config.consider_all_requests_local+ is set to true, otherwise the body is empty.
   class HostAuthorization
     ALLOWED_HOSTS_IN_DEVELOPMENT = [".localhost", IPAddr.new("0.0.0.0/0"), IPAddr.new("::/0")]
-    PORT_REGEX = /(?::\d+)?/.freeze
+    PORT_REGEX = /(?::\d+)?/ # :nodoc:
+    IPV6_HOSTNAME = /(?<host>[a-f0-9]*:[a-f0-9.:]+)/i # :nodoc:
+    IPV4_HOSTNAME = /(?<host>\d+.\d+.\d+.\d+)/i # :nodoc:
+    IPV4_HOSTNAME_WITH_PORT = /#{IPV4_HOSTNAME}#{PORT_REGEX}?/ # :nodoc:
+    IPV6_HOSTNAME_WITH_PORT = /\[#{IPV6_HOSTNAME}\]#{PORT_REGEX}?/ # :nodoc:
+    VALID_IP_HOSTNAME = Regexp.union( # :nodoc:
+      /\A#{IPV4_HOSTNAME}\z/,
+      /\A#{IPV6_HOSTNAME}\z/,
+      /\A#{IPV4_HOSTNAME_WITH_PORT}\z/,
+      /\A#{IPV6_HOSTNAME_WITH_PORT}\z/,
+    )
```

Comment:
Since we have separate expressions for no-port vs with-port, we probably want to remove the optional markers, yes?
```suggestion
    PORT_REGEX = /:\d+/ # :nodoc:
    IPV6_HOSTNAME = /(?<host>[a-f0-9]*:[a-f0-9.:]+)/i # :nodoc:
    IPV4_HOSTNAME = /(?<host>\d+.\d+.\d+.\d+)/i # :nodoc:
    IPV4_HOSTNAME_WITH_PORT = /#{IPV4_HOSTNAME}#{PORT_REGEX}/ # :nodoc:
    IPV6_HOSTNAME_WITH_PORT = /\[#{IPV6_HOSTNAME}\]#{PORT_REGEX}/ # :nodoc:
    VALID_IP_HOSTNAME = Regexp.union( # :nodoc:
      /\A#{IPV4_HOSTNAME}\z/,
      /\A#{IPV6_HOSTNAME}\z/,
      /\A#{IPV4_HOSTNAME_WITH_PORT}\z/,
      /\A#{IPV6_HOSTNAME_WITH_PORT}\z/,
    )
```

Reply:
oh yeah, of course. Now I have to remember why I have separated regexp for the non-port version.